tasks/upload.js

```diff
@@ -0,0 +1,54 @@
+const { task } = require('hardhat/config')
+const { create } = require('ipfs-http-client')
+const fs = require('fs').promises
+
+require('dotenv').config()
+
+task('upload', 'Upload directory to IPFS')
+  .addParam('file', 'The path of the file you wish to upload to IPFS')
+  .addOptionalParam('name', 'name of the NFT')
+  .addOptionalParam('desc', 'NFT description')
+  .setAction(async (taskArgs) => {
+    // create an instance of the HTTP API client
+    const ipfs = create(process.env.WRITE_GATEWAY)
+
+    let path = taskArgs.file
+
+    // fileName is the name of the .car file
+    const fileName = path.split('/').pop()
```

Comment:
could you do some check before using fileName, such as undefined and length of filename